### ISSUE TITLE
Borders wrapping task, and activities button, too narrow on GNOME 40

### ISSUE THREAD

**Issue by frandieguez** [fthx/babar#19]
Hi,

sometimes in GNOME 40 the box that encloses groups, like tasks or the activities button, is too narrow as you can see in the next image.
![image](https://user-images.githubusercontent.com/4584/112800236-ae00f500-906f-11eb-9b5c-299a3025ce0e.png)

This is weird because sometimes there is enough space and others dont. This issue can be seen more times in the activities button.

Regards.

**Comment by fthx** [comment]
Ok, the Activities "bug" make me find the culprit: reducing padding option.
Because if Activities is checked in BaBar prefs, I don't hack in any way this GS button, except through CSS...
We have to find a way to reduce between-elements padding without modifying inner padding. Don't know if it's possible.
Turning off padding reduction does solve the bug but is not what we want, I think.

**Comment by fthx** [comment]
Ok, I'm about to solve this.

**Comment by fthx** [comment]
Could you try this CSS?
I tried some hacks but did not find a padding that is good for all elements (Activities, Places, Tasks...).

```
.leftbox-reduced-padding .panel-button {
  -natural-hpadding: 6px;
  -minimum-hpadding: 6px;
}

.workspace-active-squared {
  margin-left: 10px;
  margin-right: 10px;
  background-color: rgba(204, 204, 204, 1);
  color: rgba(0, 0, 0, 1);
  border: 1px solid rgba(0, 0, 0, 0.7);
  border-radius: 2px;
}

.workspace-inactive-squared {
  margin-left: 10px;
  margin-right: 10px;
  background-color: rgba(127, 127, 127, 1);
  color: rgba(0, 0, 0, 1);
  border: 1px solid rgba(0, 0, 0, 0.7);
  border-radius: 2px;
}

.workspace-active-rounded {
  margin-left: 10px;
  margin-right: 10px;
  background-color: rgba(204, 204, 204, 1);
  color: rgba(0, 0, 0, 1);
  border: 1px solid rgba(0, 0, 0, 0.7);
  border-radius: 10px;
}

.workspace-inactive-rounded {
  margin-left: 10px;
  margin-right: 10px;
  background-color: rgba(127, 127, 127, 1);
  color: rgba(0, 0, 0, 1);
  border: 1px solid rgba(0, 0, 0, 0.7);
  border-radius: 10px;
}

.favorite {
  margin-left: 0px;
  margin-right: 0px;
  padding: 0px;
  spacing: 16px;
}

.window-box {
  margin-left: 4px;
  margin-right: 4px;
  border-radius: 0px;
  padding: 0px;
}

.window-focused {
  border-top: 2px solid rgba(204, 204, 204, 1);
  border-right: 0px hidden;
  border-bottom: 2px hidden;
  border-left: 0px hidden;
}

.window-unfocused {
  border-top: 2px solid rgba(255, 255, 255, 0);
  border-right: 0px hidden;
  border-bottom: 2px hidden;
  border-left: 0px hidden;
}

.window-hidden {
  border-top: 2px solid rgba(255, 255, 255, 0);
  border-right: 0px hidden;
  border-bottom: 2px hidden;
  border-left: 0px hidden;
}

.window-tooltip {
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: rgba(0, 0, 0, 0.7);
  color: rgba(255, 255, 255, 1);
  border: 1px solid rgba(0, 0, 0, 0.7);
  border-radius: 2px;
}

```

**Comment by frandieguez** [comment]
Seems to fix the issue for the task and favorites group, but not for the activities button

**Comment by fthx** [comment]
Ok I realized that I did a slight modification in extension.js:
include:
```
// Activities button
    	this._show_activities(DISPLAY_ACTIVITIES);
```
at the end of `enable` function, instead of the actual place.

**Comment by frandieguez** [comment]
Seems to improve a little bit but not quite what it should be

This is with babar
![image](https://user-images.githubusercontent.com/4584/113207742-474b2980-9271-11eb-81ea-b6accdb02415.png)

and this is stock
![image](https://user-images.githubusercontent.com/4584/113207826-62b63480-9271-11eb-97c4-6a96794a90ad.png)

**Comment by fthx** [comment]
As far as I understand the new frame in GS 40, its size is based on elements padding. I don't really know why.
So if we want to reduce padding, the frame will be reduced. You can disable the reduce padding option...
Or maybe there is another CSS property that does the magic.

**Comment by fthx** [comment]
Maybe give an option to change Activities label to an icon, for users like you that don't use the workspaces buttons?

**Comment by fthx** [comment]
Ok I consider this one solved.
https://github.com/fthx/babar/commit/fac033c865171ccd5ff1af7e2a783997b638c2f7

I did not succeed to do better for Activities.
But now the top frame for active task is included in box background in GS 40, it's much better.

The downside is that it reduces the optimal task icon size in GS < 40. I changed the prefs item accordingly (default to 18, but tell user that 20 is better for GS 40). The top panel has height 27px in GS < 40, but 32px in GS 40...
